From: Ubuntu Core Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Thu, 14 Feb 2019 18:59:14 -0800
Subject: disable-sem-check

# DP: Assume working semaphores, don't rely on running kernel for the check.
# DP: Assume working semaphores, don't rely on running kernel for the check.
---
 configure.ac | 34 ++++++++++++++++++++++------------
 1 file changed, 22 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4af585a..31e612d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5534,12 +5534,17 @@ AC_CACHE_CHECK([whether POSIX semaphores are enabled], [ac_cv_posix_semaphores_e
   [ac_cv_posix_semaphores_enabled=no],
   [ac_cv_posix_semaphores_enabled=yes])
 )
-AS_VAR_IF([ac_cv_posix_semaphores_enabled], [no], [
-  AC_DEFINE(
-    [POSIX_SEMAPHORES_NOT_ENABLED], [1],
-    [Define if POSIX semaphores aren't enabled on your system]
-  )
-])
+AC_MSG_RESULT($ac_cv_posix_semaphores_enabled)
+if test $ac_cv_posix_semaphores_enabled = no
+then
+  case $ac_sys_system in
+    Linux*) # assume yes, see https://launchpad.net/bugs/630511
+      ;;
+    *)
+      AC_DEFINE(POSIX_SEMAPHORES_NOT_ENABLED, 1,
+                [Define if POSIX semaphores aren't enabled on your system])
+  esac
+fi
 
 dnl Multiprocessing check for broken sem_getvalue
 AC_CACHE_CHECK([for broken sem_getvalue], [ac_cv_broken_sem_getvalue],
@@ -5571,12 +5576,17 @@ AC_CACHE_CHECK([for broken sem_getvalue], [ac_cv_broken_sem_getvalue],
   [ac_cv_broken_sem_getvalue=yes],
   [ac_cv_broken_sem_getvalue=yes])
 )
-AS_VAR_IF([ac_cv_broken_sem_getvalue], [yes], [
-  AC_DEFINE(
-    [HAVE_BROKEN_SEM_GETVALUE], [1],
-    [define to 1 if your sem_getvalue is broken.]
-  )
-])
+AC_MSG_RESULT($ac_cv_broken_sem_getvalue)
+if test $ac_cv_broken_sem_getvalue = yes
+then
+  case $ac_sys_system in
+    Linux*) # assume yes, see https://launchpad.net/bugs/630511
+      ;;
+    *)
+      AC_DEFINE(HAVE_BROKEN_SEM_GETVALUE, 1,
+      [define to 1 if your sem_getvalue is broken.])
+  esac
+fi
 
 AC_CHECK_DECLS([RTLD_LAZY, RTLD_NOW, RTLD_GLOBAL, RTLD_LOCAL, RTLD_NODELETE, RTLD_NOLOAD, RTLD_DEEPBIND, RTLD_MEMBER], [], [], [[#include <dlfcn.h>]])
 
